composer.json

```diff
@@ -74,7 +74,6 @@
     },
     "require-dev": {
         "ext-simplexml": "*",
-        "brianium/paratest": "^6.3",
```

Comment:
We have an issue with `paratest`:

On one hand we can't update `paratest` to the newest version as it requires phpunit 10.4 but we use 9.6.
On other hand we can't update to phpunit 10 as we use `phpspec/prophecy-phpunit` which is not compatible with newest phpunit. Also `symfony/phpunit-bridge` is not compatible with newest phpunit either (see https://github.com/symfony/symfony/issues/49069) yet.

I didn't want to update phpunit in this PR and decided to remove `paratest` as it seems like we don't use it in CI at all. I suppose it was used to speed up tests locally only.


Additionally I think we could get rid of `phpspec/prophecy` (in another PR) as we don't use it a lot and it could be easily replace by phpunit itself.

Reply:
I always use `paratest` locally as it speed ups quite a slow test suite.

It also finds dependencies between tests (like this one: https://github.com/infection/infection/pull/1544/files#r683828251) which directly affects on MSI (before this fix, we had incorrectly killed mutants).

So, I would like to keep it TBH.

----

> Additionally I think we could get rid of phpspec/prophecy (in another PR) as we don't use it a lot and it could be easily replace by phpunit itself.

I would better do this (if it unblocks PHPUnit 10)

> Also symfony/phpunit-bridge is not compatible with newest phpunit either (see https://github.com/symfony/symfony/issues/49069) yet.

do you know where we use it? on one of the Symfony project we got rid of it because of PHPUnit 10 incopatibility, but it wasn't used at all.

Reply:
> do you know where we use it?
everywhere :) https://github.com/infection/infection/blob/master/phpunit.xml.dist#L39

- https://github.com/infection/infection/blob/master/.github/workflows/tests.yaml#L61
- https://github.com/infection/infection/blob/master/.github/workflows/tests.yaml#L100

Reply:
> It also finds dependencies between tests (like this one: https://github.com/infection/infection/pull/1544/files#r683828251) which directly affects on MSI (before this fix, we had incorrectly killed mutants).

This is a fix for paratest as far as I can see. Am I missed smth?

Reply:
> This is a fix for paratest as far as I can see. Am I missed smth?

it's for Infection itself, and for paratest, because both libs expose `TEST_TOKEN`. 

When infection is executed on itself with `--threads=X, where X > 1`, tests are executed in parallel, and without that fix there were failing tests, which led to killed (incorrectly) mutants.

So, in other words, introducing Paratest at that time discovered that bug.

Reply:
discussed. Currently this seems like the most optimal solution. Probably, paratest will be revised later

Reply:
Paratest [`v6.11.0`](https://github.com/paratestphp/paratest/releases/tag/v6.11.0) has just been released with `Symfony:v7` support

Reply:
will update PR later today